browser-sign-in/OktaBrowserSignIn/AppDelegate.swift

```diff
@@ -15,28 +15,78 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
-
+    var configForUITests: [String: String]? {
+        let env = ProcessInfo.processInfo.environment
+        guard let oktaURL = env["OKTA_URL"], oktaURL.count > 0,
+              let clientID = env["CLIENT_ID"],
+              let redirectURI = env["REDIRECT_URI"],
+              let logoutRedirectURI = env["LOGOUT_REDIRECT_URI"] else {
+            return nil
+        }
+        return ["issuer": "\(oktaURL)/oauth2/default",
+                "clientId": clientID,
+                "redirectUri": redirectURI,
+                "logoutRedirectUri": logoutRedirectURI,
+                "scopes": "openid profile offline_access"
+        ]
+    }
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        
+        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
+            try? Keychain.Search().delete()
+        }
+        window?.rootViewController = self.getRootViewController()
```

Comment:
Don't forget that it's important to ensure the app adapts to changes in state.  For example, transitioning between sign in, to signed in, and back to sign-in when the user signs out.

Reply:
I see your point and we can use the `defaultCredentialChanged` notification observer for this i believe. 


- I was thinking through the flow and if the user signs out, the call to `WebAuthentication.signOut` is made 
- the credentials is removed, then the rest of the flow can be determined by developers. In this case, we pop to the root controller. 

- However they could be cases where you might not want to navigate back to the signin but browse as a guest.